From 0def7686c63091a8193019287ac40b5d0efad1ee Mon Sep 17 00:00:00 2001
From: Seigo Nonaka <nona@google.com>
Date: Thu, 6 Mar 2025 14:57:56 +0900
Subject: [PATCH] DO NOT MERGE - Return if the subglyph count is too large

Bug: 399065987
Test: Manually done
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:cc531278152b19d67133ba1ef1116752a35a9e6e)
Merged-In: I82c28208ecb4260cbcb6709447ae362e26c00ead
Change-Id: I82c28208ecb4260cbcb6709447ae362e26c00ead
---
 src/truetype/ttgload.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/truetype/ttgload.c b/src/truetype/ttgload.c
index 1dd319dcb..e1b873aaf 100644
--- a/src/truetype/ttgload.c
+++ b/src/truetype/ttgload.c
@@ -1955,6 +1955,11 @@
 
         /* construct an outline structure for              */
         /* communication with `TT_Vary_Apply_Glyph_Deltas' */
+        if ( gloader->current.num_subglyphs > (SHRT_MAX - 4) ) {
+            error = FT_Err_Invalid_Argument;
+            goto Exit;
+        }
+
         outline.n_points   = (short)( gloader->current.num_subglyphs + 4 );
         outline.n_contours = outline.n_points;
 
-- 
2.40.1

